src/frontend/Komponenter/Personoversikt/Infotrygdperioderoversikt.tsx

```diff
@@ -70,17 +71,99 @@ const SummertePerioder: React.FC<{ perioder: SummertPeriode[] }> = ({ perioder }
     );
 };
 
+type InfotrygdPeriodeMedFlereEndringer = InfotrygdPeriode & {
+    initialKode?: Kode;
+};
+
+/**
+ * Då det finnes flere endringer på en periode/vedtak så har noen av de høyere presedense en andre
+ * Vanligt er att man har Førstegangsvedtak/Endring/G-omregning, og sen får ett opphør, då skal opphøret vises primært
+ */
+const mapKode = (kode: Kode): number => {
+    switch (kode) {
+        case Kode.OVERTFØRT_NY_LØSNING:
+            return 5;
+        case Kode.UAKTUELL:
+            return 4;
+        case Kode.OPPHØRT:
+            return 3;
+        default:
+            return 0;
+    }
+};
+
+const formatStønadTom = (periode: InfotrygdPeriodeMedFlereEndringer): string => {
+    const stønadTom = formaterNullableMånedÅr(periode.stønadTom) as string;
+    if (periode.kode === Kode.OPPHØRT) {
+        return `${formaterNullableMånedÅr(periode.opphørsdato)} (${stønadTom})`;
+    } else {
+        return stønadTom;
+    }
+};
+
+const grupperPerioderPerVedtak = (
+    perioder: InfotrygdPeriode[]
+): { [key: string]: InfotrygdPeriode[] } =>
+    perioder.reduce((acc, periode) => {
+        const prev = acc[periode.vedtakId] || [];
+        prev.push(periode);
+        acc[periode.vedtakId] = prev;
+        return acc;
+    }, {} as { [key: string]: InfotrygdPeriode[] });
+
+/**
+ * På ett vedtak kan det finnes flere endringer (1-2), de slås sammen, men beholder initielle koden, for å kunne vise i tabellen
+ */
+const slåSammenVedtak = (
+    perioderPerVedtak: InfotrygdPeriode[][]
+): InfotrygdPeriodeMedFlereEndringer[] =>
+    perioderPerVedtak.reduce((acc: InfotrygdPeriodeMedFlereEndringer[], perioder) => {
+        if (perioder.length === 1) {
+            acc.push(perioder[0]);
+        } else {
+            const sortertePerioder = perioder.sort((a, b) =>
+                mapKode(a.kode) > mapKode(b.kode) ? -1 : 1
+            );
+            acc.push({
+                ...sortertePerioder[0],
+                initialKode: sortertePerioder[1].kode,
+            } as InfotrygdPeriodeMedFlereEndringer);
+        }
+        return acc;
+    }, [] as InfotrygdPeriodeMedFlereEndringer[]);
+
+/**
+ * Sorterer perioder, stønadFom desc, og sen på vedtakId desc, då senere vedtakId har høyere presedens
+ */
+const sortPerioder = (
+    perioder: InfotrygdPeriodeMedFlereEndringer[]
+): InfotrygdPeriodeMedFlereEndringer[] =>
+    perioder.sort((a, b) => {
+        return (
+            new Date(b.stønadFom).getTime() - new Date(a.stønadFom).getTime() ||
```

Comment:
Hva ønsker vi å oppnå her? Hvis a er nyere enn b så slår `||b.vedtakId - a.vedtakId` til. Ønsker vi kun å sjekke vedtakId hvis de er like? Isåfall tror jeg vi må justere til f.eks

`return a.stønadFom === b.stønadFom ? b.vedtakId - a.vedtakId : new Date(b.stønadFom).getTime() - new Date(a.stønadFom).getTime()`

Reply:
`||` brukes her for å gjøre noe liknende som `sortBy(..).andThen(...)`
`sortBy(stønadFom desc).andThen(vedtakId desc)`

Det kan finnes 2 vedtak med samme dato, då er det vedtakId som skal sorteres.
Dette då hvis 2 datoer er de samme, så gir `new Date(b.stønadFom).getTime() - new Date(a.stønadFom).getTime() === 0` og då slår neste sjekk inn etter `||`

Reply:
Min feil - jeg trodde `-1` var falsify, men det er den ikke - så da funker det